presto-main/src/test/java/io/prestosql/sql/analyzer/TestAnalyzer.java

```diff
@@ -1161,6 +1161,23 @@ public void testReferenceWithoutFrom()
                 .hasErrorCode(COLUMN_NOT_FOUND);
     }
 
+    @Test
+    public void testValidateFrom()
+    {
+        assertFails("SELECT * FROM (VALUES 1) a, (VALUES 2) a")
```

Comment:
I am surprised this is illegal. There is no ambiguity in the query. More precisely, intuitively, the query doesn't seem to be more ambiguous than `SELECT * FROM (VALUES 1) x(a), (VALUES 2) y(a)`.

Reply:
It is illegal by the spec. There are certain limitations on the `FROM` clause that it must not contain duplicate table names or aliases. See https://github.com/prestosql/presto/pull/1017#discussion_r335817046

Apparently, I did not even cover the limitation fully (see https://github.com/prestosql/presto/pull/1017#discussion_r335818384, the last example)